src/test/java/com/google/sps/data/FakeAvailabilityDaoTest.java

```diff
@@ -0,0 +1,271 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import static org.junit.Assert.assertEquals;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.jupiter.api.Assertions;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.junit.Test;
+
+@RunWith(JUnit4.class)
+public class FakeAvailabilityDaoTest {
+  private final FakeAvailabilityDao dao = new FakeAvailabilityDao();
+
+  private final Availability availabilityOne =
+      Availability.create(
+          "user1@mail.com",
+          new TimeRange(
+              Instant.parse("2020-07-07T12:00:00Z"), Instant.parse("2020-07-07T12:15:00Z")),
+          (long) -1,
+          true);
+
+  private final Availability availabilityTwo =
+      Availability.create(
+          "user1@mail.com",
+          new TimeRange(
+              Instant.parse("2020-07-07T15:45:00Z"), Instant.parse("2020-07-07T16:00:00Z")),
+          (long) -1,
+          false);
+
+  private final Availability availabilityThree =
+      Availability.create(
+          "user2@mail.com",
+          new TimeRange(
+              Instant.parse("2020-07-07T17:30:00Z"), Instant.parse("2020-07-07T17:45:00Z")),
+          (long) -1,
+          true);
+
+  private final Availability availabilityFour =
+      Availability.create(
+          "user1@mail.com",
+          new TimeRange(
+              Instant.parse("2020-07-07T22:30:00Z"), Instant.parse("2020-07-07T22:45:00Z")),
+          (long) -1,
+          true);
+
+  // Checks that an Availability is stored in datastore.
+  @Test
+  public void createsAvailability() {
+    dao.create(availabilityOne);
+    List<Availability> storedAvailabilities = new ArrayList<Availability>(dao.datastore.values());
+    Availability expected =
+        Availability.create(
+            "user1@mail.com",
```

Comment:
It would be clearer here if you used `availabilityOne.email()` and the other fields rather than repeating their values so that we know this is the same as the one provided above. As it is, the reader needs to realize that these are the same as those in `availabilityOne` and then go up and check that they are in fact the same.